stl/src/vector_algorithms.cpp

```diff
@@ -96,7 +96,7 @@ __declspec(noalias) void __cdecl __std_swap_ranges_trivially_swappable_noalias(
     constexpr size_t _Mask_16 = ~((static_cast<size_t>(1) << 4) - 1);
     if (_Byte_length(_First1, _Last1) >= 16 && _Use_sse2()) {
         const void* _Stop_at = _First1;
-        _Advance_bytes(_Stop_at, _Byte_length(_First1, _Last1) & _Mask_16);
+        _Advance_bytes(_Stop_at, static_cast<int>(_Byte_length(_First1, _Last1) & _Mask_16));

```

Comment:
This looks harmful for over 2 GB ranges on 64-bit platforms.

Reply:
This is already done, just implicit.